plugins/automation/db/migrate/20241219222326_add_trigger_on_field_default_value_to_topic_tags_changed_trigger.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+class AddTriggerOnFieldDefaultValueToTopicTagsChangedTrigger < ActiveRecord::Migration[7.2]
```

Comment:
Do we really need all this migration? if we go with a default value and don't change existing behavior?

Reply:
It _works_  without it but I'm not sure if it is ideal. If I have an existing Automation and then I add a default value, it still does not have that field in the table:

<img width="661" alt="Screenshot 2024-12-24 at 11 56 17" src="https://github.com/user-attachments/assets/a4b90d05-6ed7-48ad-99a1-1ffc623918a2" />

This automation was created and not migrated to the latest state, if I run `DiscourseAutomation::Field.where( automation_id: 8)` in my rails console I get the following array:

```
[#<DiscourseAutomation::Field:0x0000000122b7d7c0
  id: 15,
  automation_id: 8,
  metadata: {"value"=>"discord"},
  component: "choices",
  name: "provider",
  created_at: "2024-12-24 14:52:56.865739000 +0000",
  updated_at: "2024-12-24 14:52:56.865739000 +0000",
  target: "script">,
 #<DiscourseAutomation::Field:0x00000001257fe100
  id: 16,
  automation_id: 8,
  metadata: {"value"=>"some name"},
  component: "text",
  name: "channel_name",
  created_at: "2024-12-24 14:52:56.874948000 +0000",
  updated_at: "2024-12-24 14:52:56.874948000 +0000",
  target: "script">,
 #<DiscourseAutomation::Field:0x00000001257fdfc0
  id: 17,
  automation_id: 8,
  metadata: {},
  component: "categories",
  name: "watching_categories",
  created_at: "2024-12-24 14:52:56.882245000 +0000",
  updated_at: "2024-12-24 14:52:56.882245000 +0000",
  target: "trigger">,
 #<DiscourseAutomation::Field:0x00000001257fde80
  id: 18,
  automation_id: 8,
  metadata: {},
  component: "tags",
  name: "watching_tags",
  created_at: "2024-12-24 14:52:56.889007000 +0000",
  updated_at: "2024-12-24 14:52:56.889007000 +0000",
  target: "trigger">,
 #<DiscourseAutomation::Field:0x00000001257fdc00
  id: 19,
  automation_id: 8,
  metadata: {},
  component: "boolean",
  name: "trigger_with_pms",
  created_at: "2024-12-24 14:52:56.895092000 +0000",
  updated_at: "2024-12-24 14:52:56.895092000 +0000",
  target: "trigger">]
  created_at: "2024-12-24 14:52:56.889007000 +0000",
  updated_at: "2024-12-24 14:52:56.889007000 +0000",
  target: "trigger">]
```

It does not have the default value field, and that is ok([we only check for the other values in the automation script](https://github.com/discourse/discourse/blob/4af5591985d469fb9506538eb23c48637d4ebcdb/plugins/automation/lib/discourse_automation/event_handlers.rb#L248-L255))